libethereum/Client.cpp

```diff
@@ -91,28 +94,44 @@ void BasicGasPricer::update(BlockChain const& _bc)
 			auto bb = _bc.block(p);
 			RLP r(bb);
 			BlockReceipts brs(_bc.receipts(bi.hash()));
-			for (unsigned i = 0; i < r[1].size(); ++i)
+			size_t i = 0;
+			for (auto const& tr: r[1])
 			{
+				Transaction tx(tr.data(), CheckTransaction::None);
 				auto gu = brs.receipts[i].gasUsed();
-				dist[Transaction(r[1][i].data(), CheckTransaction::None).gasPrice()] += (unsigned)brs.receipts[i].gasUsed();
+				dist[tx.gasPrice()] += (unsigned)gu;
```

Comment:
Please, never ever c-style cast an auto variable to a POD! It might silently change into a pointer.
I would recommend not to use auto here at all.